tests/internal/testutil/testutil.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil contains utility functions used by server tests against
+// different platforms.
+package testutil
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+)
+
+const (
+	startDelay = 5 * time.Second
+	maxBackoff = time.Minute
+)
+
+// Logger writes logs (normally error messages) during retries.
+type Logger interface {
+	Logf(format string, args ...interface{})
+}
+
+// Retry keeps retrying hitting url using f with exponentially increased delay
```

Comment:
This comment needs updating:

Retry calls f until it no longer returns an error, waiting between tries.